druid/examples/readme.md

```diff
@@ -0,0 +1,216 @@
+# Examples
+
+There are several different kind of examples, some demonstrate one particular
+Druid concept, some are tools used for testing and debugging, and
+others are more complete examples of how to tie everything together.
+The later are listed separately under [showcases](##Showcases).
+
+## Anim
+```
+cargo run --example anim
+```
+This example shows how to make a simple animation using `Event::AnimFrame`.
+
+## Async Event
+```
+cargo run --example async_event
+```
+Demonstrates receiving data from some outside source, and updating the UI in response. This is similar to [blocking function](##Blocking Function) but here the data source is fully independent, and runs for the lifetime of the program.
+
+## Blocking Functions
+```
+cargo run --example blocking_functions
+```
+Sometimes you need to fetch some data from disk or from the internet,
+but you should never block the UI thread with long running operations!
+Instead you should run this task in a separate thread, and have it send
+you the data as it arrives. This is very similair to [async event](##Async Event)
+ except the event is initiated by the main thread.
+
+## Cursor
+```
+cargo run --example cursor
+```
+This example demonstrates how to set the cursor icon, and how to use a custom cursor.
+
+## Custom Widget
+```
+cargo run --example custom_widget
+```
+This shows how to use all of the methods on `PaintCtx` used for drawing on a canvas.
+You can use this to draw everything from text to images to curves.
+## Either
+```
+cargo run --example either
+```
+This example shows how to use the `Either` widget, which shows one of two children based on some predicate.
+This can be useful for loading screens, error messages, and other situations where you want to show one of
+two possible widgets.
+
+## Hello
+```
+cargo run --example hello
+```
+This shows some of the basics of druid. If you need a start of how to build an application with a text-box and some labels this is where to start.
+
+## Hello_web
+```
+cd druid/examples/hello_web
+wasm-pack build --out-dir pkg --out-name hello_web
+```
+This is an example of how to get almost any druid application can be used on the web. This is just the hello_world example but should work for all of them.
+
+## Identity
+```
+cargo run --example identity
+```
+In druid identity is used to send specific widgets commands. Instead of a command going to all the widgets, you can send them to just the one you need. This example has some colorwels and some buttons that interact with them. All of them are identical, except the identity, which makes it possible for the buttons to only affect a single colorwel.
+
+## Invalidation
+```
+cargo run --example invalidation --features="im"
+```
+A demonstration how to use debug invalidation regeons in your own widgets, including some examples of builtin widgets.
```

Comment:
regions

Reply:
😂 sometimes reading back my own texts make me laugh a bit lol. sadly the spellcheck I installed didnt seem to work in the markdown documents